tests/integration/functional/zen_stores/test_zen_store.py

```diff
@@ -0,0 +1,991 @@
+            workspace_id=client.active_workspace.id,
+            type=StackComponentType.ORCHESTRATOR,
+            name="default",
+        )
+    )[0]
+
+    with pytest.raises(IllegalOperationError):
+        store.delete_stack_component(default_artifact_store.id)
+
+    with pytest.raises(IllegalOperationError):
+        store.delete_stack_component(default_orchestrator.id)
+
+
+# .-------------------------.
+# | Stack component flavors |
+# '-------------------------'
+
+# .--------.
+# | STACKS |
+# '--------'
+
+
+def test_updating_default_stack_fails():
+    """Tests that updating the default stack is prohibited."""
+    client = Client()
+
+    default_stack = client.get_stack(DEFAULT_STACK_NAME)
+    assert default_stack.name == DEFAULT_WORKSPACE_NAME
+    stack_update = StackUpdateModel(name="axls_stack")
+    with pytest.raises(IllegalOperationError):
+        client.zen_store.update_stack(
+            stack_id=default_stack.id, stack_update=stack_update
+        )
+
+
+def test_deleting_default_stack_fails():
+    """Tests that deleting the default stack is prohibited."""
+    client = Client()
+
+    default_stack = client.get_stack(DEFAULT_STACK_NAME)
+    with pytest.raises(IllegalOperationError):
+        client.zen_store.delete_stack(default_stack.id)
+
+
+def test_get_stack_fails_with_nonexistent_stack_id():
+    """Tests getting stack fails with nonexistent stack id."""
+    client = Client()
+    store = client.zen_store
+
+    with pytest.raises(KeyError):
+        store.get_stack(uuid.uuid4())
+
+
+def test_filter_stack_succeeds():
+    """Tests getting stack."""
+    client = Client()
+    store = client.zen_store
+
+    with ComponentContext(
+        c_type=StackComponentType.ORCHESTRATOR, flavor="local", config={}
+    ) as orchestrator:
+        with ComponentContext(
+            c_type=StackComponentType.ARTIFACT_STORE, flavor="local", config={}
+        ) as artifact_store:
+            components = {
+                StackComponentType.ORCHESTRATOR: [orchestrator.id],
+                StackComponentType.ARTIFACT_STORE: [artifact_store.id],
+            }
+            with StackContext(components=components) as stack:
+                returned_stacks = store.list_stacks(
+                    StackFilterModel(name=stack.name)
+                )
+                assert returned_stacks
+
+
+def test_crud_on_stack_succeeds():
+    """Tests getting stack."""
+    client = Client()
+    store = client.zen_store
+
+    with ComponentContext(
+        c_type=StackComponentType.ORCHESTRATOR, flavor="local", config={}
+    ) as orchestrator:
+        with ComponentContext(
+            c_type=StackComponentType.ARTIFACT_STORE, flavor="local", config={}
+        ) as artifact_store:
+            components = {
+                StackComponentType.ORCHESTRATOR: [orchestrator.id],
+                StackComponentType.ARTIFACT_STORE: [artifact_store.id],
+            }
+            new_stack = StackRequestModel(
```

Comment:
StackContext should help here, no ?

Reply:
Here we are explicitely testing the crud on stacks, I feel the actual logic being tested should not be abstracted away in a context